**Olympics Dataset**

Winter and Summer Games were held in the same year up until 1992. After that, they staggered them such that Winter Games occur on a four year cycle starting with 1994, then Summer in 1996, then Winter in 1998, and so on. A common mistake people make when analyzing this data is to assume that the Summer and Winter Games have always been staggered.

### Content

The file athlete\_events.csv contains 271116 rows and 15 columns. Each row corresponds to an individual athlete competing in an individual Olympic event (athlete-events). The columns are:

1. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">ID</span> - Unique number for each athlete
2. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Name</span> - Athlete's name
3. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Sex</span> - M or F
4. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Age</span> - Integer
5. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Height</span> - In centimeters
6. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Weight</span> - In kilograms
7. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Team</span> - Team name
8. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">NOC</span> - National Olympic Committee 3-letter code
9. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Games</span> - Year and season
10. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Year</span> - Integer
11. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Season</span> - Summer or Winter
12. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">City</span> - Host city
13. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Sport</span> - Sport
14. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Event</span> - Event
15. <span style="border: 0px; font-family: inherit; font-style: inherit; font-variant: inherit; font-weight: 700; margin: 0px; padding: 0px; vertical-align: baseline;">Medal</span> - Gold, Silver, Bronze, or NA

**Importing the Data into PostgreSQL**

In [ ]:
CREATE TABLE IF NOT EXISTS OLYMPIC_HISTORY
(
	id INT,
	name VARCHAR,
	sex VARCHAR,
	age VARCHAR,
	height VARCHAR,
	weight VARCHAR,
	team VARCHAR,
	noc VARCHAR,
	games VARCHAR,
	year VARCHAR,
	season VARCHAR,
	city VARCHAR,
	sport VARCHAR,
	event VARCHAR,
	medal VARCHAR
);

CREATE TABLE IF NOT EXISTS OLYMPIC_HISTORY_NOC_REGIONS
(
	noc VARCHAR,
	region VARCHAR,
	notes VARCHAR
);

----Although I have created table as Olympics_History and Olympics_History_noc_Regions ,so further I am using this only

1. #### How many olympics games have been held?

In [9]:
Select COUNT(DISTINCT(games)) as Total_Olympic_Games
FROM olympic_history;

(1 row(s) affected)

Total execution time: 00:00:00.488

total_olympic_games 51

**2.List down all Olympics games held so far.**

In [13]:
SELECT DISTINCT(YEAR),season,city
FROM OLYMPIC_HISTORY
Order by year;

(52 row(s) affected)

Total execution time: 00:00:00.253

year season city 1896 Summer Athina 1900 Summer Paris 1904 Summer St. Louis 1906 Summer Athina 1908 Summer London 1912 Summer Stockholm 1920 Summer Antwerpen 1924 Summer Paris 1924 Winter Chamonix 1928 Summer Amsterdam 1928 Winter Sankt Moritz 1932 Winter Lake Placid 1932 Summer Los Angeles 1936 Summer Berlin 1936 Winter Garmisch-Partenkirchen 1948 Summer London 1948 Winter Sankt Moritz 1952 Summer Helsinki 1952 Winter Oslo 1956 Summer Stockholm 1956 Summer Melbourne 1956 Winter Cortina d'Ampezzo 1960 Winter Squaw Valley 1960 Summer Roma 1964 Winter Innsbruck 1964 Summer Tokyo 1968 Winter Grenoble 1968 Summer Mexico City 1972 Winter Sapporo 1972 Summer Munich 1976 Summer Montreal 1976 Winter Innsbruck 1980 Winter Lake Placid 1980 Summer Moskva 1984 Winter Sarajevo 1984 Summer Los Angeles 1988 Summer Seoul 1988 Winter Calgary 1992 Summer Barcelona 1992 Winter Albertville 1994 Winter Lillehammer 1996 Summer Atlanta 1998 Winter Nagano 2000 Summer Sydney 2002 Winter Salt Lake City 2004 Summer Athina 2006 Winter Torino 2008 Summer Beijing 2010 Winter Vancouver 2012 Summer London 2014 Winter Sochi 2016 Summer Rio de Janeiro

**3\. Mention the total no of nations who participated in each olympics game ?**

In [2]:
SELECT GAMES,Count(DISTINCT NR.REGION) Participated_Nations
from olympic_history OH
JOIN OLYMPIC_HISTORY_NOC_REGIONS AS NR USING (NOC)
GROUP BY GAMES
ORDER BY GAMES;


(51 row(s) affected)

Total execution time: 00:00:01.720

games participated_nations 1896 Summer 12 1900 Summer 31 1904 Summer 14 1906 Summer 20 1908 Summer 22 1912 Summer 29 1920 Summer 29 1924 Summer 45 1924 Winter 19 1928 Summer 46 1928 Winter 25 1932 Summer 47 1932 Winter 17 1936 Summer 49 1936 Winter 28 1948 Summer 58 1948 Winter 28 1952 Summer 66 1952 Winter 30 1956 Summer 70 1956 Winter 32 1960 Summer 82 1960 Winter 30 1964 Summer 93 1964 Winter 36 1968 Summer 110 1968 Winter 36 1972 Summer 119 1972 Winter 34 1976 Summer 90 1976 Winter 36 1980 Summer 80 1980 Winter 36 1984 Summer 138 1984 Winter 48 1988 Summer 155 1988 Winter 56 1992 Summer 167 1992 Winter 64 1994 Winter 67 1996 Summer 195 1998 Winter 72 2000 Summer 198 2002 Winter 76 2004 Summer 199 2006 Winter 78 2008 Summer 202 2010 Winter 81 2012 Summer 203 2014 Winter 88 2016 Summer 204

**4\. Which year saw the highest and lowest no of countries participating in olympics ?**

In [23]:


WITH cteasMinMaxCountries as(
SELECT games,Count(DISTINCT(noc)) as CountriesCount
FROM Olympic_History
GROUP by games)
Select DISTINCT CONCAT(
FIRST_VALUE(games) OVER (ORDER BY CountriesCount),'-',
FIRST_VALUE(CountriesCount) OVER (ORDER BY CountriesCount)) AS LowestCountries,
CONCAT(
FIRST_VALUE(games) OVER (ORDER by CountriesCount DESC),'-',
FIRST_VALUE(CountriesCount) OVER (Order by CountriesCount DESC)) as HighestCountries
FROM cteasMinMaxCountries;


(1 row(s) affected)

Total execution time: 00:00:01.300

lowestcountries highestcountries 1896 Summer-12 2016 Summer-207

**5\. Which nation has participated in all of the olympic games**

In [36]:
Select noc.region as Countries,Count(distinct games) as Total_Countries
from olympic_history oh
JOIN OLYMPIC_HISTORY_NOC_REGIONS as noc 
ON oh.noc = noc.noc
GROUP by region
Having COUNT(distinct games) = (SELECT Count(distinct games) from Olympic_History) 


(4 row(s) affected)

Total execution time: 00:00:01.690

countries total_countries France 51 Italy 51 Switzerland 51 UK 51

**6\. Identify the sport which was played in all summer olympics.**

In [75]:
 with t1 as
          	(select distinct games, sport
          	from olympic_history),
          t2 as
          	(select sport, count(1) as no_of_games
          	from t1
          	group by sport)
      select t2.*, t1.games
      from t2
      join t1 on t1.sport = t2.sport
      where t2.no_of_games = 1
      order by t1.sport;

(10 row(s) affected)

Total execution time: 00:00:00.147

sport no_of_games games Aeronautics 1 1936 Summer Basque Pelota 1 1900 Summer Cricket 1 1900 Summer Croquet 1 1900 Summer Jeu De Paume 1 1908 Summer Military Ski Patrol 1 1924 Winter Motorboating 1 1908 Summer Racquets 1 1908 Summer Roque 1 1904 Summer Rugby Sevens 1 2016 Summer

**Rank athletes based on the count of medals they've won.**

In [37]:
SELECT ID,name,COUNT(Medal) as Medal_Count,
    Rank() OVER(order by Count(medal)desc) as Medal_Rank
FROM olympic_history
WHERE Medal IN('Gold','Silver','Bronze')
GROUP by ID,name 
ORDER by Medal_Count Desc
Limit 10;



(10 row(s) affected)

Total execution time: 00:00:00.144

id name medal_count medal_rank 94406 Michael Fred Phelps, II 28 1 67046 Larysa Semenivna Latynina (Diriy-) 18 2 4198 Nikolay Yefimovich Andrianov 15 3 11951 Ole Einar Bjrndalen 13 4 74420 Edoardo Mangiarotti 13 4 109161 Borys Anfiyanovych Shakhlin 13 4 89187 Takashi Ono 13 4 23426 Natalie Anne Coughlin (-Hall) 12 8 121258 Dara Grace Torres (-Hoffman, -Minas) 12 8 87390 Paavo Johannes Nurmi 12 8

**8\. Fetch the total no of sports played in each olympic games.**

In [96]:
with t1 AS(
        Select DISTINCT games,sport
        from olympic_history),
t2 AS (
    Select games,COUNT(*) as no_of_Sports
    from t1
    GROUP by GAMES)
    SELECT * FROM t2
    ORDER by no_of_Sports DESC;



(51 row(s) affected)

Total execution time: 00:00:00.194

games no_of_sports 2000 Summer 34 2004 Summer 34 2016 Summer 34 2008 Summer 34 2012 Summer 32 1996 Summer 31 1992 Summer 29 1988 Summer 27 1920 Summer 25 1984 Summer 25 1908 Summer 24 1936 Summer 24 1972 Summer 23 1980 Summer 23 1976 Summer 23 1964 Summer 21 1924 Summer 20 1948 Summer 20 1900 Summer 20 1968 Summer 20 1952 Summer 19 1960 Summer 19 1956 Summer 19 1932 Summer 18 1904 Summer 18 1912 Summer 17 1928 Summer 17 2014 Winter 15 2006 Winter 15 2010 Winter 15 2002 Winter 15 1998 Winter 14 1906 Summer 13 1994 Winter 12 1992 Winter 12 1968 Winter 10 1984 Winter 10 1980 Winter 10 1988 Winter 10 1924 Winter 10 1964 Winter 10 1972 Winter 10 1976 Winter 10 1948 Winter 9 1896 Summer 9 1936 Winter 8 1960 Winter 8 1952 Winter 8 1928 Winter 8 1956 Winter 8 1932 Winter 7

**9\. Fetch oldest athletes to win a gold medal ?**

In [107]:
with temp AS
            (SELECT name,sex,CAST(case when age ='NA' THEN '0' ELSE age end as int) as age,
            team,games,city,sport,event,medal
            from olympic_history),
        ranking as
            (select *,rank() over(ORDER by age desc) as rnk
            from temp
            where medal = 'Gold')
    SELECT *
    from ranking
    where rnk = 1 ;


(2 row(s) affected)

Total execution time: 00:00:00.160

name sex age team games city sport event medal rnk Charles Jacobus M 64 United States 1904 Summer St. Louis Roque Roque Men's Singles Gold 1 Oscar Gomer Swahn M 64 Sweden 1912 Summer Stockholm Shooting Shooting Men's Running Target, Single Shot, Team Gold 1

**10\. Find the Ratio of male and female athletes participated in all olympic games.**

In [109]:
with t1 as (
    SELECT SEX,COUNT(*) AS CNT 
    FROM OLYMPIC_HISTORY
    GROUP BY SEX),

t2 as (
    SELECT *, row_number() OVER(order by cnt) as rn
    from t1),
min_cnt as 
        (select cnt from t2 where rn = 1),
max_cnt as 
        (select cnt from t2 where rn = 2)
select concat('1 : ', round(max_cnt.cnt::decimal/min_cnt.cnt, 2)) as ratio
    from min_cnt, max_cnt;

(1 row(s) affected)

Total execution time: 00:00:00.114

ratio 1 : 2.64

**11\. Fetch the top 5 athletes who have won the most gold medals ?**

In [124]:
with t1 as
            (select name, team, count(1) as total_gold_medals
            from olympic_history
            where medal = 'Gold'
            group by name, team
            order by total_gold_medals desc),
        t2 as
            (select *, dense_rank() over (order by total_gold_medals desc) as rnk
            from t1)
    select name, team, total_gold_medals
    from t2
    Limit 5;

(5 row(s) affected)

Total execution time: 00:00:00.120

name team total_gold_medals Michael Fred Phelps, II United States 23 Raymond Clarence "Ray" Ewry United States 10 Paavo Johannes Nurmi Finland 9 Frederick Carlton "Carl" Lewis United States 9 Larysa Semenivna Latynina (Diriy-) Soviet Union 9

**Fetch the top 5 athletes who have won the most medals (gold/silver/bronze).**

In [126]:
 with t1 as
            (select name, team, count(1) as total_medals
            from olympic_history
            where medal in ('Gold', 'Silver', 'Bronze')
            group by name, team
            order by total_medals desc),
        t2 as
            (select *, dense_rank() over (order by total_medals desc) as rnk
            from t1)
    select name, team, total_medals
    from t2
    where rnk <= 5;

(14 row(s) affected)

Total execution time: 00:00:00.165

name team total_medals Michael Fred Phelps, II United States 28 Larysa Semenivna Latynina (Diriy-) Soviet Union 18 Nikolay Yefimovich Andrianov Soviet Union 15 Ole Einar Bjrndalen Norway 13 Borys Anfiyanovych Shakhlin Soviet Union 13 Takashi Ono Japan 13 Edoardo Mangiarotti Italy 13 Sawao Kato Japan 12 Dara Grace Torres (-Hoffman, -Minas) United States 12 Aleksey Yuryevich Nemov Russia 12 Natalie Anne Coughlin (-Hall) United States 12 Paavo Johannes Nurmi Finland 12 Ryan Steven Lochte United States 12 Jennifer Elisabeth "Jenny" Thompson (-Cumpelik) United States 12

**Create categories based on the count of medals won by athletes.**

In [19]:
SELECT ID, Name, COUNT(Medal) AS Medal_Count,
    CASE
        WHEN COUNT(Medal) >= 10 THEN 'Superstar'
        WHEN COUNT(Medal) >= 5 THEN 'Star'
        WHEN COUNT(Medal) >= 1 THEN 'Participant'
        ELSE 'No Medals'
    END AS Medal_Category
FROM OLYMPIC_HISTORY
WHERE Medal IN ('Gold', 'Silver', 'Bronze')
GROUP BY ID, Name
ORDER by medal_count DESC
Limit 50;


(50 row(s) affected)

Total execution time: 00:00:00.237

id name medal_count medal_category 94406 Michael Fred Phelps, II 28 Superstar 67046 Larysa Semenivna Latynina (Diriy-) 18 Superstar 4198 Nikolay Yefimovich Andrianov 15 Superstar 89187 Takashi Ono 13 Superstar 109161 Borys Anfiyanovych Shakhlin 13 Superstar 11951 Ole Einar Bjrndalen 13 Superstar 74420 Edoardo Mangiarotti 13 Superstar 35550 Birgit Fischer-Schmidt 12 Superstar 87390 Paavo Johannes Nurmi 12 Superstar 57998 Sawao Kato 12 Superstar 70965 Ryan Steven Lochte 12 Superstar 23426 Natalie Anne Coughlin (-Hall) 12 Superstar 85286 Aleksey Yuryevich Nemov 12 Superstar 121258 Dara Grace Torres (-Hoffman, -Minas) 12 Superstar 119922 Jennifer Elisabeth "Jenny" Thompson (-Cumpelik) 12 Superstar 21402 Viktor Ivanovych Chukarin 11 Superstar 89706 Carl Townsend Osburn 11 Superstar 11642 Matthew Nicholas "Matt" Biondi 11 Superstar 18826 Vra slavsk (-Odloilov) 11 Superstar 113912 Mark Andrew Spitz 11 Superstar 124146 Franziska van Almsick 10 Superstar 39726 Aladr Gerevich (-Gerei) 10 Superstar 129726 Isabelle Regina Werth 10 Superstar 11943 Marit Bjrgen 10 Superstar 28790 Aleksandr Nikolayevich Dityatin 10 Superstar 33557 Raymond Clarence "Ray" Ewry 10 Superstar 58489 gnes Keleti-Srkny (Klein) 10 Superstar 69210 Frederick Carlton "Carl" Lewis 10 Superstar 9747 Stefania Belmondo 10 Superstar 84381 Akinori Nakayama 10 Superstar 45039 Gary Wayne Hall, Jr. 10 Superstar 5700 Polina Hryhorivna Astakhova 10 Superstar 124694 Gerard Theodor Hubert Van Innis 10 Superstar 109416 Vitaly Venediktovich Shcherbo 10 Superstar 112111 Raisa Petrovna Smetanina 10 Superstar 96173 Aleksandr Vladimirovich Popov 9 Star 120087 Ian James Thorpe 9 Star 127501 Mikhail Yakovlevich Voronin 9 Star 109597 Martin Joseph Sheridan 9 Star 126257 Maria Valentina Vezzali 9 Star 106296 Heikki Ilmari Savolainen 9 Star 90007 Merlene Joyce Ottey-Page 9 Star 132791 Lyubov Ivanovna Yegorova 9 Star 122726 Lyudmila Ivanovna Turishcheva (-Borzova) 9 Star 28751 Ursula "Uschi" Disl 9 Star 117045 Alfred Gomer Albert "Alf" Swahn 9 Star 58702 Eizo Kenmotsu 9 Star 34551 Allyson Michelle Felix 9 Star 55692 Leisel Marie Jones 9 Star 92566 Claudia Pechstein 9 Star

**How does the distribution of medals differ between male and female athletes?**

In [34]:
SELECT Sex,Medal,COUNT(*) AS Medal_Count
FROM OLYMPIC_HISTORY
WHERE Medal IN('Gold','Silver','Bronze')
GROUP BY Sex,Medal


(6 row(s) affected)

Total execution time: 00:00:00.202

sex medal medal_count F Bronze 3771 F Gold 3747 F Silver 3735 M Bronze 9524 M Gold 9625 M Silver 9381

**Certainly! Let's refine these insights using the given information about the Olympics dataset:**

**\-Total Olympic Games Held:**

<span style="color: var(--vscode-foreground);">There have been a total of 51 Olympic Games held from 1896 to 2016.</span>  

**\-Nations' Participation:**

The highest participation of nations in a single Olympic event was 199 in the 2004 Summer Olympics, while the lowest was 12 in the 1896 Summer Olympics.

Specifically, France, Italy, Switzerland, and the UK were among the consistent participating countries across various Olympic events.

**\-Most Decorated Athletes:**

Michael Fred Phelps II holds the record for the most Olympic medals, winning 28 medals.

Larysa Semenivna Latynina follows closely with 18 medals, while Urszula Kielan won one medal and ranked 6997 among athletes.

**\-Sports in Specific Olympic Games:**

The number of sports was consistent at 34 in the 2000, 2004, 2016, and 2008 Summer Olympics.

**\-Oldest Gold Medalists:**

Charles Jacobus and Oscar Gomer Swahn were the oldest athletes to win a gold medal. Charles won in the 1904 Summer Olympics, while Oscar achieved this feat in the 1912 Summer Olympics.

**\-Gender Participation Ratio:**

The overall ratio of male to female participation across all Olympic Games stands at approximately 1:2.64.

**\-Top 5 Athletes by Gold Medals:**

The top five athletes with the most gold medals are Michael Fred Phelps, Raymond Clarence, Paavo Johannes Nurmi, Frederick Carlton, and Larysa Semenivna Latynina.

**These insights summarize various aspects of the Olympics dataset, including historical participation, outstanding athletes, nations' involvement, gender representation, and exceptional achievements in sports.**